.github/workflows/buildtest.yaml

```diff
@@ -11,36 +11,47 @@ jobs:
   test:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-    - uses: actions/cache@v2
-      with:
-        path: |
-          ~/go/pkg/mod
-          ~/go/bin
-          ~/.cache
-        key: livekit-server
-    - uses: shogo82148/actions-setup-redis@v1.10.3
-      with:
-        redis-version: '6.x'
-        auto-start: true
-    - run: redis-cli ping
-
-    - name: Set up Go
-      uses: actions/setup-go@v2
-      with:
-        go-version: 1.17
-
-    - name: Download Go modules
-      run: go mod download
-
-    - name: Mage Build
-      uses: magefile/mage-action@v1
-      with:
-        version: latest
-        args: build
-
-    - name: Mage Test
-      uses: magefile/mage-action@v1
-      with:
-        version: latest
-        args: testall
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/go/pkg/mod
+            ~/go/bin
+            ~/.cache
+          key: livekit-server
+      - uses: shogo82148/actions-setup-redis@v1.10.3
+        with:
+          redis-version: '6.x'
+          auto-start: true
+      - run: redis-cli ping
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17
+
+      - name: Set up gotestfmt
+        run: go install github.com/haveyoudebuggedit/gotestfmt/v2/cmd/gotestfmt@latest
+
+      - name: Download Go modules
+        run: go mod download
+
+      - name: Mage Build
+        uses: magefile/mage-action@v1
+        with:
+          version: latest
+          args: build
+
+      - name: Test
+        run: |
+          set -euo pipefail
+          go test -json -v ./... 2>&1 | tee /tmp/gotest.log | gotestfmt
+
+      # Upload the original go test log as an artifact for later review.
+      - name: Upload test log
+        uses: actions/upload-artifact@v2
+        if: always()
```

Comment:
why the `if`?

Reply:
I got this from their example yaml.. not totally sure either.

Reply:
The if is there to make sure the test log gets uploaded even if the test fails in the previous step. You would want to have the raw log available in all cases. (Maintainer of `gotestfmt` here. :) )

Reply:
ah, that's great to know. Thanks @janosdebugs for your work on `gotestfmt`!